### ISSUE TITLE
💅 noUnusedFunctionParameters false positives in svelte files that use `#snippet`

### ISSUE THREAD

**Issue by dyc3** [biomejs/biome#10245]
### Environment information

<details>

```bash

CLI:
  Version:                      2.4.14
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:                    unset
  BIOME_LOG_PREFIX_NAME:             unset
  BIOME_LOG_LEVEL:                   unset
  BIOME_LOG_KIND:                    unset
  BIOME_CONFIG_PATH:                 unset
  BIOME_THREADS:                     unset
  BIOME_WATCHER_KIND:                unset
  BIOME_WATCHER_POLLING_INTERVAL:    unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v24.3.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         bun/1.3.3

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.jsonc
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true
  HTML full support enabled:    true

Workspace:
  Open Documents:               0

```

</details>

### Rule name

noUnusedFunctionParameters

### Playground link

https://biomejs.dev/playground/?lintRules=noUnusedFunctionParameters&tab=formatter&pane=Diagnostics&code=PABzAGMAcgBpAHAAdAAgAGwAYQBuAGcAPQAiAHQAcwAiAD4ACgBpAG0AcABvAHIAdAAgAHQAeQBwAGUAIAB7ACAAVwBpAHQAaABvAHUAdABDAGgAaQBsAGQAIAB9ACAAZgByAG8AbQAgACIAYgBpAHQAcwAtAHUAaQAiADsACgBpAG0AcABvAHIAdAAgACoAIABhAHMAIABGAG8AcgBtAFAAcgBpAG0AaQB0AGkAdgBlACAAZgByAG8AbQAgACIAZgBvAHIAbQBzAG4AYQBwACIAOwAKAGkAbQBwAG8AcgB0ACAAewAgAEwAYQBiAGUAbAAgAH0AIABmAHIAbwBtACAAIgAkAGwAaQBiAC8AYwBvAG0AcABvAG4AZQBuAHQAcwAvAHUAaQAvAGwAYQBiAGUAbAAvAGkAbgBkAGUAeAAuAGoAcwAiADsACgBpAG0AcABvAHIAdAAgAHsAIABjAG4AIAB9ACAAZgByAG8AbQAgACIAJABsAGkAYgAvAHUAdABpAGwAcwAuAGoAcwAiADsACgAKAGwAZQB0ACAAewAKAAkAcgBlAGYAIAA9ACAAJABiAGkAbgBkAGEAYgBsAGUAKABuAHUAbABsACkALAAKAAkAYwBoAGkAbABkAHIAZQBuACwACgAJAGMAbABhAHMAcwA6ACAAYwBsAGEAcwBzAE4AYQBtAGUALAAKAAkALgAuAC4AcgBlAHMAdABQAHIAbwBwAHMACgB9ADoAIABXAGkAdABoAG8AdQB0AEMAaABpAGwAZAA8AEYAbwByAG0AUAByAGkAbQBpAHQAaQB2AGUALgBMAGEAYgBlAGwAUAByAG8AcABzAD4AIAA9ACAAJABwAHIAbwBwAHMAKAApADsACgA8AC8AcwBjAHIAaQBwAHQAPgAKAAoAPABGAG8AcgBtAFAAcgBpAG0AaQB0AGkAdgBlAC4ATABhAGIAZQBsACAAewAuAC4ALgByAGUAcwB0AFAAcgBvAHAAcwB9ACAAYgBpAG4AZAA6AHIAZQBmAD4ACgAJAHsAIwBzAG4AaQBwAHAAZQB0ACAAYwBoAGkAbABkACgAewAgAHAAcgBvAHAAcwAgAH0AKQB9AAoACQAJADwATABhAGIAZQBsACAAewAuAC4ALgBwAHIAbwBwAHMAfQAgAGMAbABhAHMAcwA9AHsAYwBuACgAIgBkAGEAdABhAC0AZgBzAC0AZQByAHIAbwByADoAdABlAHgAdAAtAGQAZQBzAHQAcgB1AGMAdABpAHYAZQAiACwAIABjAGwAYQBzAHMATgBhAG0AZQApAH0APgAKAAkACQAJAHsAQAByAGUAbgBkAGUAcgAgAGMAaABpAGwAZAByAGUAbgA%2FAC4AKAApAH0ACgAJAAkAPAAvAEwAYQBiAGUAbAA%2BAAoACQB7AC8AcwBuAGkAcABwAGUAdAB9AAoAPAAvAEYAbwByAG0AUAByAGkAbQBpAHQAaQB2AGUALgBMAGEAYgBlAGwAPgAKAA%3D%3D&language=svelte&ruleDomains.react=recommended

### Expected result

It should not flag this parameter because it is used in the snippet body.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
